tqp/topic_queue_poller.py

```diff
@@ -219,26 +219,31 @@ def ensure_queue(self):
         queue_arn = queue.attributes["QueueArn"]
 
         sns = boto3.resource("sns")
-        policies = []
+        topic_arns = []
 
         for topic_name in self.handlers.keys():
             topic = sns.create_topic(Name=topic_name)
-            policies.append(
-                {
-                    "Sid": "sns",
-                    "Effect": "Allow",
-                    "Principal": {"AWS": "*"},
-                    "Action": "SQS:SendMessage",
-                    "Resource": queue_arn,
-                    "Condition": {"ArnEquals": {"aws:SourceArn": topic.arn,},},
-                }
-            )
-
             topic.subscribe(Protocol="sqs", Endpoint=queue_arn)
 
+            topic_arns.append(topic.arn)
+
         queue.set_attributes(
             Attributes=_jsonify_dictionary(
-                {"Policy": {"Version": "2012-10-17", "Statement": policies},}
+                {
+                    "Policy": {
+                        "Version": "2012-10-17",
+                        "Statement": {
+                            "Sid": "sns",
+                            "Effect": "Allow",
+                            "Principal": {"AWS": "*"},
+                            "Action": "SQS:SendMessage",
+                            "Resource": queue_arn,
+                            "Condition": {
+                                "ArnEquals": {"aws:SourceArn": topic_arns}
```

Comment:
Are you sure this works? See here: https://docs.aws.amazon.com/sns/latest/dg/sns-access-policy-language-api-permissions-reference.html

`Each policy must cover only a single topic (when writing a policy, don't include statements that cover different topics)`

Reply:
(It's possible that that's not relevant, since this is an SQS policy.)

Reply:
Here's what I get from the UI:

![image](https://user-images.githubusercontent.com/697647/86255366-0492de80-bb85-11ea-9b7b-66ef6734adfc.png)

maybe you need

```suggestion
                                "ArnEquals": {"aws:SourceArn": ",".join(topic_arns)}
```

? just double check either way

Reply:
@emmilco I think that specific docco is only applicable to SNS.

That said, I also can't find any docs suggesting that an array will work here.

From https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_ARN:

> Case-sensitive matching of the ARN. Each of the six colon-delimited components of the ARN is checked separately and each can include a multi-character match wildcard (*) or a single-character match wildcard (?). These behave identically.

Can we do wildcard matching here instead?